app/models/user_role.rb

```diff
@@ -139,10 +139,32 @@ def discourse_user_group
   end
 
   def can_user_read?(user)
-    # A user can view a role if:
-    # 1. the role belongs to a non-hidden group, or
-    # 2. the user has read-access to that group.
-    !group.is_hidden || user&.has_permission?(:can_read_groups, group.id)
+    return true unless group.is_hidden
+    return false if user.nil?
+    if group.banned_competitors?
+      if is_active? # The actively banned competitors can be viewed by following set of people:
+        (
+          user.wdc_team? || # 1. WDC
+          user.ethics_committee? || # 2. WEC
+          user.board_member? || # 3. Board
+          user.staff_delegate? || # 4. Staff Delegates
+          user.weat_team? || # 5. WEAT
+          user.results_team? || # 6. WRT
+          user.admin? # 7. Admin
+        )
+      else
+        (
+          user.wdc_team? || # 1. WDC
+          user.ethics_committee? || # 2. WEC
+          user.board_member? || # 3. Board
+          user.weat_team? || # 4. WEAT
+          user.results_team? || # 5. WRT
+          user.admin? # 6. Admin
```

Comment:
This and the or list in the if block can be combined by moving to a method. But the thing is the list can change in future - like we can reach a stage were one user can read past and not active, while another user can read active but not past. To handle these situations and to make the function more readable, I went for this approach. Please don't ask to move to user.rb :-)